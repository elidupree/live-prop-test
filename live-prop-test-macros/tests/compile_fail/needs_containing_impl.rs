use live_prop_test::live_prop_test;

struct Struct;
trait Trait {
  type Associated;
}
impl Trait for Struct {
  type Associated = ();
}

impl Struct {
  #[live_prop_test(postcondition = "false")]
  fn constructor() -> Self {
    Struct
  }

  #[live_prop_test(postcondition = "false")]
  fn associated_constructor() -> <Self as Trait>::Associated {}

  #[live_prop_test(postcondition = "false")]
  fn by_ref_method(&self) {}

  #[live_prop_test(postcondition = "false")]
  fn by_value_method(self) {}
}

trait AssociatedConstructor: Trait {
  #[live_prop_test(postcondition = "false")]
  fn associated_constructor() -> Self::Associated;
}

fn main() {}
