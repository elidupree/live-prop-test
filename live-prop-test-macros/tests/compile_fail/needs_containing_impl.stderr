error: when using live-prop-test on methods inside an `impl`, you must also put the #[live_prop_test] attribute on the containing impl
  --> $DIR/needs_containing_impl.rs:13:23
   |
13 |   fn constructor() -> Self {
   |                       ^^^^

error: when using live-prop-test on methods inside an `impl`, you must also put the #[live_prop_test] attribute on the containing impl
  --> $DIR/needs_containing_impl.rs:18:35
   |
18 |   fn associated_constructor() -> <Self as Trait>::Associated {}
   |                                   ^^^^

error: when using live-prop-test on methods inside an `impl`, you must also put the #[live_prop_test] attribute on the containing impl
  --> $DIR/needs_containing_impl.rs:21:20
   |
21 |   fn by_ref_method(&self) {}
   |                    ^

error: when using live-prop-test on methods inside an `impl`, you must also put the #[live_prop_test] attribute on the containing impl
  --> $DIR/needs_containing_impl.rs:24:22
   |
24 |   fn by_value_method(self) {}
   |                      ^^^^

error: when using live-prop-test on methods inside an `impl`, you must also put the #[live_prop_test] attribute on the containing impl
  --> $DIR/needs_containing_impl.rs:29:34
   |
29 |   fn associated_constructor() -> Self::Associated;
   |                                  ^^^^
